step-capstone/src/components/Map.js

```diff
@@ -6,37 +6,219 @@ import React, { createRef } from 'react'
  * center: {lat:0.0, lng:0.0} - coordinates for center of map
  */
 
+class MapComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      finalizedMarkers: new Map(),
+      unfinalizedMarkers: new Map(),
+      geoPaths: []
+    }
+    this.googleMapRef = createRef();
+    this.googleMap = undefined
+  }
 
-class Map extends React.Component {
-  
   componentDidMount() {
+    console.log("MOUNTING");
     const loadGoogleMapScript = document.createElement('script');
     loadGoogleMapScript.src =
-     'https://maps.googleapis.com/maps/api/js?key='+process.env.REACT_APP_API_KEY+'&libraries=place';
-    
+      'https://maps.googleapis.com/maps/api/js?key=' + process.env.REACT_APP_API_KEY + '&libraries=place';
+
     window.document.body.appendChild(loadGoogleMapScript);
 
     loadGoogleMapScript.addEventListener('load', () => {
       this.googleMap = this.createMap();
+      let bounds = this.drawMap()
+      this.googleMap.fitBounds(bounds);
     });
   }
 
-  googleMapRef = createRef();
+  /*
+   * Updates state with any changes in props
+   */
+  componentDidUpdate(prevProps) {
+    if (prevProps !== this.props) {
+      this.clearMap();
+      this.drawMap();
+    }
+  }
+
   createMap() {
-    new window.google.maps.Map(this.googleMapRef.current, {
+    return new window.google.maps.Map(this.googleMapRef.current, {
       zoom: this.props.zoom,
       center: this.props.center,
     })
   }
 
+  addMarker(coordinates, type) {
+    // TODO: firebase will provide coordinates as a GeoPoint --> convert to {lat, lng}
+    // { lat: coordinates.lat(), lng: coordinates.lng() }
+
+    var iconUrl;
+    if (type === "flight") {
+      iconUrl = "http://maps.google.com/mapfiles/ms/icons/blue-dot.png"
+    } else if (type === "event") {
+      iconUrl = "http://maps.google.com/mapfiles/ms/icons/green-dot.png"
+    } else {
+      iconUrl = "http://maps.google.com/mapfiles/ms/icons/red-dot.png"
+    }
+
+    var newMarker = new window.google.maps.Marker({
+      position: coordinates,
+      map: this.googleMap,
+      animation: window.google.maps.Animation.DROP,
+      icon: { url: iconUrl }
+    })
+
+    // zoom to marker when clicked
+    newMarker.addListener('click', () => {
+      this.googleMap.setZoom(15);
+      this.googleMap.setCenter(newMarker.getPosition());
+
+      // TODO: this.props.onMarkerClicked(id) --> notifies parent that marker clicked
```

Comment:
I was thinking when the marker is clicked, the travelobject is scrolled to / highlighted in the itinerary. Needed later because it is not implemented yet / not yet merged with itinerary component. I put it there as a placeholder when it would may eventually be needed.